pkg/store/proxy_heap.go

```diff
@@ -462,6 +463,49 @@ func newLazyRespSet(
 					seriesStats.Count(resp.GetSeries())
 				}
 
+				if resp.GetCompressedSeries() != nil {
```

Comment:
I wonder if we can hide this behind some abstraction that knows whether to decompress the response or not. Maybe `GetSeries()` itself can do this if we pass in the hints?

Reply:
Makes sense. I forgot that we need to wait until the end to get the table.